core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -311,14 +319,27 @@ public void updateScanIteratorOption(String iteratorName, String key, String val
   }
 
   private class IterEnv implements IteratorEnvironment {
+    private final Supplier<ServiceEnvironment> serviceEnvironment;
```

Comment:
What prevents us from just using the implementation from `super.getPluginEnv()` and `super.getServiceEnv()`?

Reply:
`IteratorEnvironment` doesn't have any implementation for `getServiceEnv()` or `getPluginEnv()`. Or are you suggesting changing `IteratorEnvironment`?